lib/Backends/NNPI/NNPIOptions.h

```diff
@@ -18,166 +18,309 @@
+  }
+
+  virtual llvm::StringRef getOptionsName() const override {
+    return "Backend Options";
+  };
 };
 
-/// This class holds all environment variable knobs for the NNPI compilation.
 class NNPICompilationOptions : public NNPIOptions {
 public:
-  /// Compilation paramters used as defaults (if exists) that may be overriden
-  /// by environment variables.
-  NNPICompilationOptions(const std::map<std::string, std::string> *parameters);
-  /// Dump the status of all variables.
-  virtual std::string dumpStatus() const override;
-  /// Lists the supported options (name=>description).
-  static llvm::StringMap<std::string> getSupportedOptions();
-  /// Compile to ICE-T if true, ICE-Ref otherwise - this variable cab be
-  /// verridden by inferOnDevice_.
-  bool useIceT;
-  /// Enable execution on device (useIceT_ and !inferOnDevice_ will compile but
-  /// not execute inference).
-  bool inferOnDevice;
+  /// Compile for HW (ignored if InferOnDevice is defined).
+  DECLARE_NNPI_OPTION(useIceT, bool, "UseIceT",
+                      "Compile for HW (ignored if InferOnDevice is defined).",
+                      "USE_ICE_T", "0");
+  /// Enable execution on device (if true, will also force compilation for HW
+  /// and ignore the UseIceT option).
+  DECLARE_NNPI_OPTION(inferOnDevice, bool, "InferOnDevice",
+                      "Enable execution on device (if true, will also force "
+                      "compilation for HW and ignore the UseIceT option).",
+                      "USE_INF_API", "0");
   /// Setting this to true will log the status of all variables at backend
   /// creation.
-  bool showVars;
+  DECLARE_NNPI_OPTION(showVars, bool, "ShowVars",
+                      "Setting this to true will log the status of all "
+                      "variables at backend creation.",
+                      "NNPI_SHOW_VARS",
+#ifdef NDEBUG
+                      "0"
+#else
+                      "1"
+#endif
+  );
+  /// Setting this variable will control the logging level (0 for debug, 1
+  /// assert, 2 info, 3 warning, 4 error, 5 critical ).
+  DECLARE_NNPI_OPTION(
+      compilationLogLevel, int, "CompilationLogLevel",
+      "Sets the compilation logging level (0-6). 0=Debug, 1=Assert, 2=Info, "
+      "3=Warning, 4=Error, 5=Critical, 6=User.",
+      "NNPI_LOG_LEVEL",
+#ifdef NDEBUG
+      "4"
+#else
+      "0"
+#endif
+  );
   /// Setting this variable will save the compilation output to the filename
   /// specified.
-  std::string compiledFile;
-  /// Setting this variable will force compilation to the to use no more than
-  /// the amount of ice cores (1-12) - default 0.
-  int iceCores;
+  DECLARE_NNPI_OPTION(compiledFile, std::string, "CompiledFile",
+                      "Sets a file name to save the compilation output to the "
+                      "filename specified.",
+                      "ICE_T_FILE", "");
+  /// Force compilation with maximum amount of ice cores, -1 for unlimited.
```

Comment:
Curious -- what does unlimited mean here, as opposed to 12 as the actual max?

Reply:
We did not want to define a number limit since future version may support a different number of cores.